src/pages/SignUp/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Link, useNavigate } from 'react-router-dom';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useSelector, useDispatch } from 'react-redux';
+import { signUpSchema } from '../../schemas';
+import {
+  Button, InputField, SelectInput, PhoneSection, Title,
+} from '../../components/common';
+import { genderOptions } from '../../constants/genderOptions';
+import { SignUpInputsType } from '../../types/userInputsTypes';
+import { signUp } from '../../state/actions/userActions';
+import { requestErrorsSelector, statusSelector } from '../../state/reducers/userReducer';
+import { RequestStatus } from '../../constants/requestStatus';
+
+function SignUp() {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const status = useSelector(statusSelector);
+  const requestErrors = useSelector(requestErrorsSelector);
+
+  const isRequestLoading = status === RequestStatus.PENDING;
+
+  const errorsExist = Object.entries(requestErrors).length > 0;
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(signUpSchema) });
+
+  const onSubmit: SubmitHandler<SignUpInputsType> = async (data: SignUpInputsType) => {
+    try {
+      await dispatch(signUp(data)).unwrap();
+      navigate('/login');
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  return (
+    <section className="flex h-full">
+      <div className="text-center bg-white half-section">
+        <Title text="Sign Up" />
+        {errorsExist
+          && <span className="error-message">{requestErrors?.full_messages[0]}</span>}
```

Comment:
Could you create a function to extract errors that come from api? ,  looks like `requestErrors?.full_messages[0]`  is not the only one here, a function that extract everything what you need could be really helpfull :)

Reply:
Probably you need an extra validation here, what if i get more than one error from the api and those are related to different fields?